### PR TITLE
Fixed hardcoded antispam and antivirus host addresses

### LINKED ISSUE #978: Should `antispam` be hardcoded?
I'm in a middle of setting up my own mailu instances, however I'm using namespaced container names (i.e. `mailu-admin`, `mailu-front` etc.). Everything works pretty well so far, however I have some problems with imap and antispam. After some investigation I noticed, that files `core/dovecot/conf/bin/ham` and `core/dovecot/conf/bin/spam` have antispam host hardcoded instead of using `HOST_ANTISPAM`. 

I'm happy to provide PR, but firstly want to be sure - is it in fact oversight, or is this by design?

### PR DESCRIPTION
**Pull request by ajgon**
## What type of PR?

enchancement

## What does this PR do?

Adds configurable parameters `HOST_ANTISPAM` and `HOST_ANTIVIRUS` for `core/dovecot` and `services/rspamd`, instead of using hardcoded container names.

### Related issue(s)
- closes #978

### CODE REVIEW

**Comment by kaiyou** [comment]
This looks ok globally, but you seem to be using yaml parsing in the bash scripts, while they could be using the environment variables directly. Also, you compile those scripts to `/etc/dovecot`, while the sieve bin directory remains untouched in `/conf/bin`.

I believe your best option here is to use environment variables directly in the bin scripts. You will need to update the process environment in the start script though, since it currently runs on a copy of environ.

**Comment by ajgon** [comment]
@kaiyou Fair point, I updated the PR.

**Comment by kaiyou** [comment]
bors try

**Comment by kaiyou** [comment]
I am adding a priority label since it is approved and I want it merged in 1.7.

**Comment by kaiyou** [comment]
@micw, since you worked on hosts, could you comment a bit?

**Comment by micw** [comment]
@kaiyou This is great, it replaces my https://github.com/Mailu/Mailu/pull/1110 and fixes another hardcoded server too.

**Comment by muhlemmer** [comment]
I'm blocking until we fix and merge #940. That should be done until tomorrow.

**Comment by kaiyou** [comment]
#940 should be merged any minute now

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ v1.6.1 - unreleased
 - Enhancement: Stop using static assets, but build them using Webpack
 - Enhancement: Create an Authentication Token with IPv6 address restriction ([#829](https://github.com/Mailu/Mailu/issues/829))
 - Enhancement: Missing wildcard option in alias flask command ([#869](https://github.com/Mailu/Mailu/issues/869))
+- Enhancement: Fixed hardcoded antispam and antivirus host addresses ([#978](https://github.com/Mailu/Mailu/issues/978))
```
Comment (muhlemmer) [inline CHANGELOG.md]: This should be in towncrier: https://mailu.io/master/contributors/guide.html#changelog

File: core/dovecot/conf/bin/ham [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
-tee >(rspamc -h antispam:11334 -P mailu learn_ham /dev/stdin) \
-    | rspamc -h antispam:11334 -P mailu -f 13 fuzzy_add /dev/stdin
\ No newline at end of file
+tee >(rspamc -h ${HOST_ANTISPAM:-antispam:11334} -P mailu learn_ham /dev/stdin) \
+    | rspamc -h ${HOST_ANTISPAM:-antispam:11334} -P mailu -f 13 fuzzy_add /dev/stdin
```
Comment (muhlemmer) [inline core/dovecot/conf/bin/ham]: We should now use `ANTISPAM_ADDRESS`, and should be defined in `start.py`

File: core/dovecot/conf/bin/spam [hunk 3]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
-tee >(rspamc -h antispam:11334 -P mailu learn_spam /dev/stdin) \
-    >(rspamc -h antispam:11334 -P mailu -f 11 fuzzy_add /dev/stdin)
\ No newline at end of file
+tee >(rspamc -h ${HOST_ANTISPAM:-antispam:11334} -P mailu learn_spam /dev/stdin) \
+    >(rspamc -h ${HOST_ANTISPAM:-antispam:11334} -P mailu -f 11 fuzzy_add /dev/stdin)
```
Comment (muhlemmer) [inline core/dovecot/conf/bin/spam]: See `core/dovecot/conf/bin/ham`

File: services/rspamd/conf/antivirus.conf [hunk 4]
```diff
@@ -3,6 +3,6 @@ clamav {
   attachments_only = true;
   symbol = "CLAM_VIRUS";
   type = "clamav";
-  servers = "antivirus:3310";
+  servers = "{{ HOST_ANTIVIRUS }}";
```
Comment (muhlemmer) [inline services/rspamd/conf/antivirus.conf]: Should become `ANTIVIRUS_ADDRESS`

File: services/rspamd/start.py [hunk 5]
```diff
@@ -10,6 +10,7 @@
 
 # Actual startup script
 os.environ["FRONT_ADDRESS"] = system.resolve_address(os.environ.get("FRONT_ADDRESS", "front"))
+os.environ["HOST_ANTIVIRUS"] = system.resolve_address(os.environ.get("HOST_ANTIVIRUS", "antivirus:3310"))
```
Comment (muhlemmer) [inline services/rspamd/start.py]:
````
os.environ["ANTIVIRUS_ADDRESS"] = ....
````

**Pull request merged**